Jenkinsfile

```diff
@@ -37,325 +38,552 @@ pipeline {
+            }
           }
-          agent { label 'armv8' }
+          agent { label 'mac' }
           steps {
             script {
-              def releaseBuild = load ".jenkinsci/release-build.groovy"
-              releaseBuild.doReleaseBuild()
+              def macDebugBuild = load ".jenkinsci/mac-debug-build.groovy"
+              def testSelect = load ".jenkinsci/test-launcher.groovy"
+              macDebugBuild.doTestStep(testSelect.chooseTestType())
             }
           }
-          post {
-            always {
-              script {
-                post = load ".jenkinsci/linux-post-step.groovy"
-                post.linuxPostStep()
+        }
+      }
+    }
+    stage('Post-Coverage') {
+      when {
+        anyOf {
+          expression { params.Coverage }  // by request
+          allOf {
+            expression { return env.CHANGE_ID }
+            not { expression { return env.GIT_PREVIOUS_COMMIT } }
+          }
+          allOf {
+            expression { return env.CHANGE_ID != null}
+            expression { return env.GIT_PREVIOUS_COMMIT != null }
+            expression { return env.CHANGE_TARGET ==~ /(master|develop|trunk)/ }
+            expression { return params.Merge_PR }
+          }
+          allOf {
+            expression { return params.BUILD_TYPE == 'Debug' }
+            expression { return env.GIT_LOCAL_BRANCH ==~ /master/ }
+          }
+        }
+      }
+      parallel {
+        stage('lcov_cobertura') {
+          steps {
+            script {
+              def cov_platform = ''
+              if (params.Linux) {
+                cov_platform = 'x86_64_aws_cov'
+              }
+              if (!params.Linux && params.MacOS) {
+                cov_platform = 'mac'
+              }
+              if (!params.Linux && !params.MacOS && params.ARMv8) {
+                cov_platform = 'armv8'
+              }
+              else if (!params.Linux && !params.MacOS && !params.ARMv8) {
+                cov_platform = 'armv7'
+              }
+              node(cov_platform) {
+                if (cov_platform == 'mac') {
+                  coverage = load '.jenkinsci/mac-debug-build.groovy'
+                }
+                else {
+                  coverage = load '.jenkinsci/debug-build.groovy'
+                }
+                  coverage.doPostCoverageCoberturaStep()
               }
             }
           }
         }
-        stage('MacOS') {
+        stage('sonarqube') {
+          when {
+            anyOf {
+              expression { params.Coverage }  // by request
+              allOf {
+                expression { return env.CHANGE_ID }
+                not { expression { return env.GIT_PREVIOUS_COMMIT } }
+              }
+              allOf {
+                expression { return env.CHANGE_ID != null}
+                expression { return env.GIT_PREVIOUS_COMMIT != null }
+                expression { return params.Merge_PR }
```

Comment:
`expression { return env.CHANGE_TARGET ==~ /(master|develop|trunk)/ }` is not required here, why?

Reply:
actually we can remove the whole section `when` in `parallel` of stage `Post-coverage` as it is needless after `when` in `Post-coverage`